plugin/acl/acl.go

```diff
@@ -0,0 +1,103 @@
+package acl
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/acl/filter"
+	"github.com/coredns/coredns/plugin/metrics"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+type acl struct {
+	Next plugin.Handler
+
+	Rules []Rule
+}
+
+// Rule defines a list of Zones and some ACL policies which will be
+// enforced on them.
+type Rule struct {
+	Zones    []string
+	Policies []Policy
+}
+
+// Policy defines the ACL policy for DNS queries.
+// A policy performs the specified action (block/allow) on all DNS queries
+// matched by source IP or QTYPE.
+type Policy struct {
+	action string
+	qtype  uint16
+	filter filter.Filter
+}
+
+const (
+	// ALLOW allows authorized queries to recurse.
+	ALLOW string = "allow"
```

Comment:
these can be just (untyped) numbers.
~~~
const (
 Allow = iota
 Block
)
~~~

Reply:
Sorry but I think they have to be string since I'm using them to compare with directives.